saq/queue/postgres.py

```diff
@@ -484,7 +484,29 @@ async def iter_jobs(
 
     async def abort(self, job: Job, error: str, ttl: float = 5) -> None:
         job.error = error
-        await self.update(job, status=Status.ABORTING)
+
+        async with self.pool.connection() as conn, conn.cursor() as cursor:
+            await cursor.execute(
+                SQL(
+                    dedent(
+                        """
+                        SELECT status
+                        FROM {jobs_table}
+                        WHERE key = %(key)s
+                        FOR UPDATE
+                        """
+                    )
+                ).format(
+                    jobs_table=self.jobs_table,
+                ),
+                {
+                    "key": job.key,
+                },
+            )
+            result = await cursor.fetchone()
+            await self.update(job, status=Status.ABORTING, connection=conn)
```

Comment:
why isn't this in the same block? should it be

if result == Queued:
  await job.finish(Aborted):
  else:
await self.update(job, ...)